HW_7/logger.py

```diff
@@ -0,0 +1,11 @@
+from datetime import datetime
+
+
+def log(func):
+    def wapper(*args):
+        with open('log.csv', 'a', encoding='utf-8') as log_file:
+            log_file.write(f'{datetime.now()} - запущена функция' 
+                            '{func.__name__} {"с аргументами" if args else "без аргументов"}' 
+                            '({func.__doc__})\n')
```

Comment:
Если вы переносите подобный образом строку, нужно перед каждым переносом и ставить букву `f.`
```suggestion
            log_file.write(f'{datetime.now()} - запущена функция' 
                            f'{func.__name__} {"с аргументами" if args else "без аргументов"}' 
                            f'({func.__doc__})\n')
```
Иначе ваш лог файл будет выглядеть таким образом.
![image](https://user-images.githubusercontent.com/49741194/208309348-8b931633-a7ea-456a-8038-06f65147ac07.png)
Поэтому меняла снова вопрос, вы точно запускали ваше приложение и проверяли его функциональность?

Reply:
Кроме того, у вас ни одна функция, которую вы декорируете декораторам `@log` не имеет строки документации, но в самом декораторе прописано извлечение этой строки.